formbricks/formbricks
# 473 refactor template page

## What does this PR do?

Converts client components used in Templates page to Server components, with the aim to eliminate use of useSWR, which reduces the number of roundtrips between server and client

**Summary of changes**
- Built a wrapper component around the client components in page.tsx,making page a server component. This is the place where requests happen
- Build typesafe generic services to get product, environment
- Add loading.tsx in the same folder as page.tsx, to ensure loading animation displayed for loading-state of the server component(waiting for data)



Fixes #473 



## Type of change




## How should this be tested?



- Check for same functionality for added endpoints
- Check logs for cached requests


## Checklist
